build_datasets.py

```diff
@@ -1,247 +1,289 @@
+    parser.add_argument("--en-output-type", "--en", required=True,
+                        choices=EN_OUTPUT_CHOICES)
+
+    # Paths
+    parser.add_argument("-i", "--input-mfa-dir", default='covost_mfa/data',
+                        help="Directory containing MFA JSON files")
+    parser.add_argument("-o", "--processed-output-dir", default='processed_datasets',
+                        help="Parent directory for output data")
+    parser.add_argument("--covost-dir", default='./covost_tsv',
+                        help="Directory containing covost TSV files")
+    parser.add_argument("--durations-path", default='durations_freq_all.pkl',
+                        help="Pickle file containing dictionary of durations"
+                        " and corresponding frequencies")
+    parser.add_argument("--bpe-de", default='data/training/de_codes_10k',
+                        help="BPE codes for de side")
+    parser.add_argument("--bpe-en", default='data/training/en_codes_10k_mfa',
+                        help="BPE codes for en side")
+    parser.add_argument("--force-redo", "-f", action='store_true',
+                        help="Redo datasets even if the output directory already exists")
+    parser.add_argument("--write-segments-to-file", action='store_true',
+                        help="Write unnoise and unbinned segment durations to a separate file")
+
+    # Other arguments
+    parser.add_argument("--upsampling", type=int, default=1,
+                        help="Upsample examples by this factor (for noisy outputs)")
+    parser.add_argument("--noise-std", type=float, default=0.0,
+                        help="Standard deviation for noise added to durations")
+    parser.add_argument("--num-bins", type=int, default=100,
+                        help="Number of bins. 0 means no binning.")
+
+    args = parser.parse_args()
+
+    # Read data
+    train_tsv, dev_tsv, test_tsv = load_tsv(args.covost_dir)
+    codes_de = codecs.open(args.bpe_de, encoding='utf-8')
     bpe_de = BPE(codes_de)
-    codes_en = codecs.open("./en_codes_10k_mfa", encoding='utf-8')
+    codes_en = codecs.open(args.bpe_en, encoding='utf-8')
     bpe_en = BPE(codes_en)
 
-    durations_path = './durations_freq_all.pkl'
-    if os.path.exists(durations_path):
-        with open(durations_path, 'rb') as f:
-            durations_pkl = pickle.load(f)
-            print("loaded durations' freq!")
+    assert os.path.exists(args.durations_path), \
+        "Run get_durations_frequencies.py first to get the dictionary of durations" \
+        " and how many times each is observed in our data!"
+    with open(args.durations_path, 'rb') as f:
+        logging.info("Loading durations' frequencies")
+        durations_pkl = pickle.load(f)
+    if not os.path.exists(args.processed_output_dir):
+        os.makedirs(args.processed_output_dir)
+
+    output_path = os.path.join(args.processed_output_dir, args.de_output_type)
+    if args.num_bins == 0:
```

Comment:
Is there a case where we do not bin at all? I would add a warning here that Binning is turned off.

Reply:
I included the option when I thought we might use it, but didn't actually end up using it. Added a warning as you suggest.